heaps/min_heap.py

```diff
@@ -57,20 +67,51 @@ def heap_up(self, index):
             property is reestablished.
             
             This could be **very** helpful for the add method.
-            Time complexity: ?
-            Space complexity: ?
+            Time complexity: ? O(log n)
+            Space complexity: ? O(1)
         """
-        pass
+        parent_index = int((index - 1) / 2)
+        current_key = self.store[index].key 
+        parent_key = self.store[parent_index].key 
 
+        while current_key < parent_key:
+            self.swap(index, parent_index)
+            index = parent_index
+            parent_index = int((index - 1) / 2)
+            current_key = self.store[index].key
+            parent_key = self.store[parent_index].key 
+
+    def has_left_child(self, index):
+        left_child_index = int((2 * index) + 1)
+        if left_child_index <= len(self.store) - 1:
+            return True
+        return False
```

Comment:
👀 Performing a boolean check (if condition) then returning a boolean is an anti-pattern. We can return the value of the condition directly.

```py
        return left_child_index <= len(self.store) - 1
```